Data/ByteString/Short/Internal.hs

```diff
@@ -482,6 +487,56 @@ toShortIO (BS fptr len) = do
     touchForeignPtr fptr
     ShortByteString <$> stToIO (unsafeFreezeByteArray mba)
 
+-- | A simple wrapper around 'fromShort' that wraps the strict 'ByteString' as
+-- a one-chunk lazy 'LBS.ByteString'.
+lazyFromShort :: ShortByteString -> LBS.ByteString
+lazyFromShort = LBS.fromStrict . fromShort
+
+-- | /O(n)/. Convert a lazy 'LBS.ByteString' into a 'ShortByteString'.
+--
+-- This makes a copy, so does not retain the input string.  Naturally, best
+-- used only with sufficiently short lazy ByteStrings.
+--
+lazyToShort :: LBS.ByteString -> ShortByteString
+lazyToShort LBS.Empty = empty
+lazyToShort (LBS.Chunk c@(BS _ len) cs) =
+    unsafeDupablePerformIO $ do
+        mba0 <- stToIO (newByteArray len)
+        mba <- lazyToShortIO mba0 0 len c cs
+        ShortByteString <$> stToIO (unsafeFreezeByteArray mba)
+
+-- | Append first and any subsequent chunks of lazy bytestring to the provided
+-- buffer, resizing it if required.  The return value is the final buffer.
+-- Arithmetic is done with care, to avoid undetected integer addition
+-- wrap-around.  While nobody should be copying multi-gigabyte lazy bytestrings
+-- to "short" bytestrings, we should error out if possible, rather than corrupt
+-- memory..
+lazyToShortIO :: MutableByteArray RealWorld      -- ^ result buffer
+              -> Int                             -- ^ space already used
+              -> Int                             -- ^ total space allocated
+              -> ByteString                      -- ^ initial strict chunk
+              -> LBS.ByteString                  -- ^ remaining chunks
+              -> IO (MutableByteArray RealWorld) -- ^ possibly resized result
+lazyToShortIO mba used size c@(BS fptr len) cs
```

Comment:
Would it be better not to pass `size`, but rather extract it every time with `getSizeofMutableByteArray#`? Admittedly this would require IO before any pattern-matching begins...

Reply:
This is a purely internal helper function, we could even move it into a `where` clause, to really make sure there are no unexpected calls. So I don't think such caution is warranted, I was much more concerned with handling integer arithmetic overflow, which I believe is handled correctly.